docs/guides/src/main/server/configuring-hostname.adoc

```diff
@@ -0,0 +1,180 @@
+
+By default, the URLs for the administration console are also based on the incoming request. However, you can set a specific host or base URL if you want
+to restrict access to the administration console using a specific URL. Similarly to how you set the frontend URLs, you can use the `hostname-admin` and `hostname-admin-url` options to achieve that.
+
+Most of the time, it should be enough to set the `hostname-admin` option in order to change only the *host* of the administration console URLs:
+
+<@kc.start parameters="--hostname-admin=<host>"/>
+
+However, if you want to set not only the host but also a scheme, port, and path, you can set the `hostname-admin-url` option:
+
+<@kc.start parameters="--hostname-admin-url=<scheme>://<host>:<port>/<path>"/>
+
+If you don't set these options and the administration console is accessed using a hostname other than what is set to the frontend URLs, you
+might get an error from the server telling you that the redirect URI used by the console is invalid. In this case, you should update the
+`security-admin-console` client to add a valid redirect URI based on the hostname you want the administration console to be accessible.
+
+Note that the `hostname-admin` and `hostname-admin-url` are mutually exclusive.
+
+To reduce attack surface, the administration endpoints for Keycloak and the Admin Console should not be publicly accessible.
+Therefore, you can secure them by using a reverse proxy.
+For more information about which paths to expose using a reverse proxy, see the <@links.server id="reverseproxy"/> Guide.
+
+== Accessing Keycloak in production mode using HTTP
+
+When a `hostname` is set and the server is running in production mode, all the URLs generated by the server are going to use the `HTTPS` scheme. If you are not setting up TLS you might run into issues because some URLs generated by the server won't work.
+
+Keycloak follows the "secure by design" principle, so it is absolutely not recommended to access Keycloak without proper transport encryption, as this opens up multiple attack vectors.
+
+Nevertheless, there are environments where Keycloak is deployed behind a proxy/load balancer that terminates TLS completely and the internal requests are done using the unencrypted HTTP protocol.
+
+To be able to work with Keycloak using HTTP for these environments, there is the hidden configuration option `hostname-strict-https=<true/false>`. This option is set to `true` by default for the production mode, and `false` for the development mode.
+
+When you need to access Keycloak using HTTP in production mode, for example when you use `proxy=edge` and you want to access the administration console internally using HTTP, you have to set `hostname-strict-https=false`, otherwise a blank page will show up.
+
+Keep in mind the recommended approach is to always use HTTPS, and this still is true for external clients.
+
+== Example Scenarios
```

Comment:
I would say they were confusing as they are now but after changing the examples to show realm claims from the discovery document, I think it adds value so that people can understand how the options interfere with the URLs created by the server.